lib/array_intersection.rb

```diff
@@ -1,4 +1,17 @@
 def intersection(list1, list2)
-  raise NotImplementedError, "Intersection not implemented"
+  intersecting_numbers = []
+  hash = {}
 
+  list1.map do |number|
+    hash[number] = false
+  end
+
+  list2.each do |number|
+    if hash.keys.include? number
+      hash[number] = true #guess I don't really need this step
+      intersecting_numbers << number
+    end
+  end
```

Comment:
Since both `false` and `nil` are falsy, I suggest assigning `hash[number]` to true in the 1st loop.

```suggestion
  list1.map do |number|
    hash[number] = true
  end

  list2.each do |number|
    if hash[number]
      intersecting_numbers << number
    end
  end
```